packages/core/src/browser/preferences/preference-service.ts

```diff
@@ -131,6 +143,25 @@ export class PreferenceServiceImpl implements PreferenceService, FrontendApplica
         return value !== null && value !== undefined ? value : defaultValue;
     }
 
+    async set(preferenceName: string, value: string, scope: PreferenceScope = PreferenceScope.Workspace): Promise<void> {
+        let preferenceProvider: PreferenceProvider;
```

Comment:
Please consider to extract a method: `const preferenceProvider = getPreferenceProvider(scope)`.
It should simplify this method. `getPreferenceProvider` will be simpler as well - one can get rid of a local variable and replace `break`s with `return`s.

Reply:
Reworked to `PreferenceProviderFactory`